src/app/features/public-form/public-form.component.ts

```diff
@@ -93,10 +98,24 @@ export class PublicFormComponent<E extends Entity> implements OnInit {
     this.formConfig = publicForms.find(
       (form: PublicFormConfig) => form.route === id || form.getId(true) === id,
     );
+    if (!this.formConfig) {
+      this.publicFormNotFound = {
+        title: $localize`Public Form Not Found`,
+        error: $localize`The form you are looking for is either unavailable or doesn't exist. Please check the link and try again.`,
+      };
+      return;
+    }
 
     this.entityType = this.entities.get(
       this.formConfig.entity,
     ) as EntityConstructor<E>;
+    if (this.ability.cannot("create", this.entityType)) {
+      this.publicFormNotFound = {
```

Comment:
rename to "error"?

Reply:
and to simplify the component code, maybe it's easier to just add a `error: "not_found" | "no_permissions"` and move the texts to the html template file